src/db.c

```
@@ -1070,6 +1070,104 @@ void moveCommand(client *c) {
     addReply(c,shared.cone);
 }
 
+void copyCommand(client *c) {
+    robj *o;
+    redisDb *src, *dst;
+    int srcid;
+    long long dbid, expire;
+    int j, replace = 0, delete = 0;
+    
+    if (c->argc >= 6){
+        addReply(c,shared.syntaxerr);
+        return;
+    }
+    
+    /* Obtain source and target DB pointers 
+     * Default target DB is the same as the source DB 
+     * Parse the REPLACE option and targetDB option. */
+    src = c->db;
+    srcid = c->db->id;
+    dbid = c->db->id;
+    for (j = 3; j < c->argc; j++) {
+        if (!strcasecmp(c->argv[j]->ptr,"replace")) {
+            replace = 1;
+        } else if (getLongLongFromObject(c->argv[j],&dbid) == C_ERR ||
+            dbid < INT_MIN || dbid > INT_MAX ||
+            selectDb(c,dbid) == C_ERR)
+        {
+            addReply(c,shared.outofrangeerr);
+            return;
+        }
```

Would prefer to error instead of silently ignoring bad arguments.